.github/workflows/publish-prerelease.yml

```diff
@@ -16,98 +17,30 @@ jobs:
-            Start-Sleep -s 1
-        }
-
-    - name: Setup .NET 8
+    - name: Setup .NET SDK
       uses: actions/setup-dotnet@v4
       with:
         dotnet-version: 8.0.x
 
-    # Determine new version
-    - name: Install GitVersion
-      uses: gittools/actions/gitversion/setup@v0
-      with:
-        versionSpec: '5.x'
-
-    - name: Determine Version
-      id: gitversion
-      uses: gittools/actions/gitversion/execute@v0
-      with:
-        useConfigFile: true
-        configFilePath: gitversion.yml
-
     - name: Restore dependencies
       run: dotnet restore
 
-    - name: Build
-      run: dotnet build --no-restore -c ${{ env.Configuration }} /p:Version=${{ steps.gitversion.outputs.FullSemVer }}
+    - name: Build solution [Release]
+      run: dotnet build --no-restore -c Release
 
-    - name: Run tests
-      run: dotnet test --no-build -c ${{ env.Configuration }} -p:CollectCoverage=true -p:CoverletOutputFormat=opencover -p:CoverletOutput=out/.coverage/
+    - name: Pack solution [Release]
+      run: dotnet pack --no-build -c Release -o dist/packages
 
-    - name: Upload coverage reports to Codecov
-      uses: codecov/codecov-action@v4
-      with:
-        files: out/.coverage/*.opencover.xml
-        token: ${{ secrets.CODECOV_TOKEN }} # required
-    
-    - name: Create nuget packages
-      run: dotnet pack --no-build -c ${{ env.Configuration }} -o dist/packages /p:PackageVersion=${{ steps.gitversion.outputs.FullSemVer }}
-
-    - uses: actions/upload-artifact@v4
+    - name: Upload artifacts
+      uses: actions/upload-artifact@v4
       with:
         name: nuget-packages
         path: dist/packages
 
-  deploy-github:
-    runs-on: 'ubuntu-latest'
-    environment: 'github'
-    needs: build
-    steps:
-      - uses: actions/download-artifact@v4
-        with:
-          name: nuget-packages
-          path: dist
-
-      - name: Publish Nuget to GitHub registry
-        run: dotnet nuget push ./dist/*.nupkg -k ${GITHUB_TOKEN} -s https://nuget.pkg.github.com/ArieGato/index.json --skip-duplicate --no-symbols
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-
-  deploy-nuget:
-    runs-on: 'ubuntu-latest'
-    environment: 'nuget'
-    needs: deploy-github
-    steps:
-      - uses: actions/download-artifact@v4
-        with:
-          name: nuget-packages
-          path: dist
+    - name: Publish Nuget to GitHub registry
+      run: dotnet nuget push ./dist/*.nupkg --api-key ${{ secrets.GITHUB_TOKEN }} --source https://nuget.pkg.github.com/ArieGato/index.json
 
-      - name: Publish Nuget to nuget.org registry
-        run: dotnet nuget push ./dist/*.nupkg --api-key ${NUGET_API_KEY} --source https://api.nuget.org/v3/index.json --skip-duplicate --no-symbols
-        env:
-          NUGET_API_KEY: ${{ secrets.NUGET_API_KEY }}
+    - name: Publish Nuget to nuget.org registry
```

Comment:
Do we want to push a package to nuget.org for every change to master? So when Dependabot bumps a package, a new package is pushed.

Reply:
Completely not. I just preserved your worflow with some cosmetic changes. In all other repos I tend to use `release` event to publish nuget packages to nuget.org. I wanted to remove this step in further PR but if you want I can remove it here.

Reply:
Because the jobs deploy-github and deploy-nuget are now gone we have no control anymore. If we can publish a prerelease version to nuget.org with the publish-release.yml, you can remove the step here.

Reply:
I mean that I would not publish preview version on nuget.org regardless of workflows organization.

Reply:
So, only publish prereleases to github feed. And only releases to nuget.

Reply:
yes

Reply:
then go right ahead to remove the step to publish to nuget.org

Reply:
already done